src/generator.cs

```diff
@@ -3004,7 +3004,33 @@ void GenerateTrampolinesForQueue (TrampolineInfo [] queue)
 			print ("}} /* class {0} */", ti.NativeInvokerName);
 		}
 	}
-	
+
+	// We need to check against the user using just UIKit (and friends) in the FieldAttribute
+	// so we need to reflect the libraries contained in our Constants class and do the mapping
+	// we will return the system library path if found
+	bool IsNotSystemLibrary (string library_name)
+	{
+		string library_path = null;
+		return IsNotSystemLibrary (library_name, ref library_path);
+	}
+
+	bool IsNotSystemLibrary (string library_name, ref string library_path)
+	{
+		var libSuffixedName = $"{library_name}Library";
+		var constType = typeof (
+#if XAMCORE_2_0
+		XamCore.ObjCRuntime.Constants);
+#else
+		XamCore.Constants);
+#endif
+		var field = constType.GetFields (BindingFlags.Public | BindingFlags.Static).FirstOrDefault (f => f.Name == libSuffixedName);
+		if (field == null)
+			return true;
+
+		library_path = (string) field.GetValue (null);
+		return false;
```

Comment:
I believe we can replace:

```csharp
var field = constType.GetFields (BindingFlags.Public | BindingFlags.Static).FirstOrDefault (f => f.Name == libSuffixedName);
if (field == null)
    return true;
 
 library_path = (string) field.GetValue (null);
 return false;
```

With the following:

```csharp
var field = constType.GetFields (BindingFlags.Public | BindingFlags.Static).FirstOrDefault (f => f.Name == libSuffixedName);

library_path = (string) field?.GetValue (null);
return library_path == null;
```

In the above code, if field is null the null-propagating will return (string) null, which is valid. If not, we do call GetValue (null) and get the library_path, then return that the library_path is null. It is important to keep the (string) cast to ensure that we do get the GetValue returned value casted.

Also, regarding the name of this method, in the .Net framework most of the methods that return a bool and set an out param or a ref one follow the pattern Try* maybe we just use just a pattern in this method. TryGetLibraryPath for example.